lib/instrumentation/trace-util.js

```diff
@@ -45,6 +47,31 @@ exports.getTraceContext = (traceIdSource, req) => {
         });
       }
 
+      case X_AMZN_TRACE_ID_HEADER: {
+        let traceId, parentSpanId;
+
+        const split = value.split(";");
+        for (const s of split) {
+          const [name, value] = s.split("=");
+          if (name === "Root") {
+            traceId = value;
+          } else if (name === "Parent") {
+            parentSpanId = value;
+          }
+        }
+
+        if (!traceId) {
+          // if we didn't even get a 'Root=' clause, bail.
+          return {};
+        }
+
```

Comment:
one thing here.  @maplebed beeline-go PoC did something like this here:

```suggestion
if (!parentId) {
  parentId = traceId;
}
```

should 100% do that here.

Reply:
Yep - I like this and think we should do it everywhere. Seems to make sense.